apitools/base/py/transfer.py

```diff
@@ -339,14 +341,39 @@ def __SetRangeHeader(self, request, start, end=None):
         else:
             request.headers['range'] = 'bytes=%d-%d' % (start, end)
 
-    def __GetChunk(self, start, end=None, additional_headers=None):
+    def __ComputeEndByte(self, start, end=None, use_chunks=True):
+        """Compute the last byte to fetch for this request.
+
+        Note that this is potentially confusing in several ways:
+          * the value for the last byte is 0-based, eg "fetch 10 bytes
+            from the beginning" would return 9 here.
+          * if we have no information about size, and don't want to
+            use the chunksize, we'll return None.
+
+        See the tests for more examples.
+
+        Args:
+          start: byte to start at.
+          end: (int or None, default: None) Suggested last byte.
+          use_chunks: (bool, default: True) If False, ignore self.chunksize.
+
+        Returns:
+          Last byte to use in a Range header, or None.
```

Comment:
This has some potentially strange interaction with negative start ranges.  Is it intended for use in that case?